src/vec.zig

```diff
@@ -80,6 +80,18 @@ pub fn rotateZ(self: anytype, angle: @typeInfo(@TypeOf(self)).vector.child) @Typ
 	};
 }
 
+pub fn rotate2d(self: anytype, angle: @typeInfo(@TypeOf(self)).vector.child, center: @TypeOf(self)) @TypeOf(self) {
+	if(@typeInfo(@TypeOf(self)).vector.len != 2) @compileError("Only available for vectors of length 2.");
+
+	const sin = @sin(angle);
+	const cos = @cos(angle);
+
+	return @TypeOf(self){
+		cos*(self[0] - center[0]) - sin*(self[1] - center[1]),
+		sin*(self[0] - center[0]) + cos*(self[1] - center[1]),
+	} + center;
+}
+
```

Comment:
Will optimizer extract `self[i] - center[i]` that subtraction to make it vector optimized?

To have it guaranteed I think we can do:

```suggestion
pub fn rotate2d(self: anytype, angle: @typeInfo(@TypeOf(self)).vector.child, center: @TypeOf(self)) @TypeOf(self) {
	if(@typeInfo(@TypeOf(self)).vector.len != 2) @compileError("Only available for vectors of length 2.");

	const sin = @sin(angle);
	const cos = @cos(angle);
	const pos = self - center;

	return @TypeOf(self){
		cos*(pos[0]) - sin*(pos[1]),
		sin*(pos[0]) + cos*(pos[1]),
	} + center;
}

```